sklearn/decomposition/_nmf.py

```diff
@@ -803,35 +914,39 @@ def _fit_multiplicative_update(X, W, H, beta_loss='frobenius',
     previous_error = error_at_init
 
     H_sum, HHt, XHt = None, None, None
-    for n_iter in range(1, max_iter + 1):
-        # update W
-        # H_sum, HHt and XHt are saved and reused if not update_H
-        delta_W, H_sum, HHt, XHt = _multiplicative_update_w(
-            X, W, H, beta_loss, l1_reg_W, l2_reg_W, gamma,
-            H_sum, HHt, XHt, update_H)
-        W *= delta_W
-
-        # necessary for stability with beta_loss < 1
-        if beta_loss < 1:
-            W[W < np.finfo(np.float64).eps] = 0.
-
-        # update H
-        if update_H:
-            delta_H = _multiplicative_update_h(X, W, H, beta_loss, l1_reg_H,
-                                               l2_reg_H, gamma)
-            H *= delta_H
-
-            # These values will be recomputed since H changed
-            H_sum, HHt, XHt = None, None, None
 
+    for n_iter in range(0, max_iter):
+        for iter_offset, slice in enumerate(
```

Comment:
nitpick. It think it makes the code easier to follow if we rename `slice` into `batch`, or define `this_X = X[slice]`.